src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlDateTimeMethodTranslator.cs

```diff
@@ -42,66 +41,66 @@ public class NpgsqlDateAddTranslator : IMethodCallTranslator
             //{ typeof(DateTimeOffset).GetRuntimeMethod(nameof(DateTimeOffset.AddMilliseconds), new[] { typeof(double) }), "milliseconds" }
         };
 
-        /// <summary>
-        /// The backend version to target.
-        /// </summary>
-        [NotNull] readonly Version _postgresVersion;
+        readonly ISqlExpressionFactory _sqlExpressionFactory;
+        readonly RelationalTypeMapping _intervalMapping;
+        readonly RelationalTypeMapping _textMapping;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="NpgsqlDateAddTranslator"/> class.
+        /// Initializes a new instance of the <see cref="Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal.NpgsqlDateTimeMethodTranslator"/> class.
         /// </summary>
-        /// <param name="postgresVersion">The backend version to target.</param>
-        public NpgsqlDateAddTranslator([CanBeNull] Version postgresVersion)
-            => _postgresVersion = postgresVersion ?? MinimumSupportedVersion;
+        public NpgsqlDateTimeMethodTranslator(
+            ISqlExpressionFactory sqlExpressionFactory,
+            IRelationalTypeMappingSource typeMappingSource)
+        {
+            _sqlExpressionFactory = sqlExpressionFactory;
+            _intervalMapping = typeMappingSource.FindMapping("interval");
+            _textMapping = typeMappingSource.FindMapping("text");
+        }
 
         /// <inheritdoc />
-        public virtual Expression Translate(MethodCallExpression methodCallExpression, IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+        public SqlExpression Translate(SqlExpression instance, MethodInfo method, IList<SqlExpression> arguments)
         {
-            // This translation is only supported for PostgreSQL 9.4 or higher.
-            if (_postgresVersion < MinimumSupportedVersion)
+            if (!MethodInfoDatePartMapping.TryGetValue(method, out var datePart))
```

Comment:
Yes - this is actually a bit interesting.

In 2.2 we had client evaluation, so we specifically refrained from generating SQL that isn't compatible with older versions of PostgreSQL; this way the expression worked, even if it wasn't translated to the server.

In 3.0 that doesn't seem to make any sense anymore, because what doesn't get translated will fail in any case. So might as well translate, and on older versions the database will error. I don't see any real value in not translating and getting an EF error as opposed to a PG error...